### PR TITLE
Refactor DataKek, data now contain list of dictionary records

### PR DESCRIPTION
**Pull request by maruschin**
I think it's better to cast DataFrame to list of dicts with using pandas functions.

### CODE REVIEW

File: kekas/data.py [hunk 1]
```diff
@@ -13,15 +13,13 @@ def __init__(self,
                  df: pd.DataFrame,
                  reader_fn: Callable,
                  transforms: Optional[Compose] = None) -> None:
-        # transform df to list
-        self.data = list(df.values)  # for multiprocessing
-        self.df_columns = df.columns
+        # transform df to list of dict records
+        self.data = df.to_dict("records")
```
Comment (belskikh) [inline kekas/data.py]: `df.to_dict` works VERY slow when the amount of entries is large
after I realized it, I switched to `list()`
Maybe, there is another way?

Comment (maruschin) [reply kekas/data.py]: Yes, we can use `iloc` and then get dict from series object:
```
df.iloc[ind].to_dict()
```

Redid PR.

Comment (belskikh) [reply kekas/data.py]: Thanks!
Did you test it with dataloader with num_workers>1 (I mean in multiprocessing regime)?
I think I had problems with a similar approach in the past

Comment (maruschin) [reply kekas/data.py]: Yes, I tested it with multiprocess, with num_workers==6

**Pull request merged**